src/plots/cartesian/layout_defaults.js

```diff
@@ -299,7 +301,15 @@ module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, fullData) {
         axLayoutIn = layoutIn[axName];
         axLayoutOut = layoutOut[axName];
 
-        handleConstraintDefaults(axLayoutIn, axLayoutOut, coerce, allAxisIds, layoutOut);
+        var scaleanchorDflt = null;
+        if(axLetter === 'y' && !axLayoutIn.hasOwnProperty('scaleanchor') && !yaMustNotScaleanchor[axName] && yaMayScaleanchor[axName]) {
```

Comment:
This line is also rather long. So it would be nice to have it on multiple lines.

Reply:
That's 4 lines instead of 2 but I'll make the change :)

Reply:
Done in d71c3b345d2eecb7b3087c383f00f5f102b9c2f6

Reply:
Non-blocking (to me :smile: ) but I think you forgot to commit this line.

Reply:
Nice catch! Done in 2053c583d6ca1cc2dffe4632738dd120abf67988